client/my-sites/earn/ads/form-settings.jsx

```diff
@@ -17,76 +16,103 @@ import FormTextInput from 'calypso/components/forms/form-text-input';
 import { getSelectedSite, getSelectedSiteId } from 'calypso/state/ui/selectors';
-import { dismissWordAdsSuccess } from 'calypso/state/wordads/approve/actions';
 import { saveWordadsSettings } from 'calypso/state/wordads/settings/actions';
 
-class AdsFormSettings extends Component {
-	static propTypes = {
-		site: PropTypes.object.isRequired,
-		markChanged: PropTypes.func.isRequired,
-		markSaved: PropTypes.func.isRequired,
-	};
-
-	state = {};
-
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( { wordadsSettings, site } ) {
-		if ( ( isEmpty( this.state ) && wordadsSettings ) || site?.ID !== this.props.site?.ID ) {
-			this.setState( {
-				...this.defaultSettings(),
+const AdsFormSettings = () => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const [ settings, setSettings ] = useState( {} );
+	const [ isChanged, setIsChanged ] = useState( false );
+
+	const site = useSelector( ( state ) => getSelectedSite( state ) );
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const siteUrl = useSelector( ( state ) => getSiteUrl( state, siteId ) );
+	const siteIsJetpack = useSelector( ( state ) => isJetpackSite( state, siteId ) );
+	const isSavingSettings = useSelector( ( state ) => isSavingWordadsSettings( state, siteId ) );
+	const wordadsSettings = useSelector( ( state ) => getWordadsSettings( state, siteId ) );
+	const widgetsUrl = useSelector( ( state ) => getCustomizerUrl( state, siteId, 'widgets' ) );
+
+	const isLoading = ! wordadsSettings || isSavingSettings;
+	const isWordAds = site?.options?.wordads;
+
+	useEffect( () => {
+		// Set default settings if settings are empty
+		const settingsAreEmptyAndWordadsSettingsExist =
+			isEmpty( settings ) && ! isEmpty( wordadsSettings );
+
+		if ( settingsAreEmptyAndWordadsSettingsExist ) {
+			setSettings( {
+				...defaultSettings(),
 				...wordadsSettings,
 			} );
 		}
-	}
+	}, [ settings, wordadsSettings ] );
+
+	useEffect( () => {
+		// siteId has changed, so we reset settings
+		// This code is here to directly mimic code that
+		// was in UNSAFE_componentWillReceiveProps
+		setSettings( {
+			...defaultSettings(),
+			...wordadsSettings,
+		} );
+
+		// Only run on siteId change
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ siteId ] );
 
-	handleChange = ( event ) => {
+	function handleChange( event ) {
 		const name = event.currentTarget.name;
 		const value = event.currentTarget.value;
 
-		this.setState( {
+		setSettings( ( prevState ) => ( {
+			...prevState,
 			[ name ]: value,
-		} );
-	};
+		} ) );
+	}
 
-	handleToggle = ( event ) => {
+	function handleToggle( event ) {
 		const name = event.currentTarget.name;
 
-		this.setState( {
-			[ name ]: ! this.state[ name ],
-		} );
-	};
+		setSettings( ( prevState ) => ( {
+			...prevState,
+			[ name ]: ! settings[ name ],
+		} ) );
+	}
```

Comment:
In the old code, there was a 'state' object that held all settings. I've renamed that to settings. 

In several of the onChange callbacks, there is a small difference in logic in the new code. For a big refactor like this, I like to keep logic the same. I'm a bit bothers this change was needed. The prior code

```
this.setState( {
			[ name ]: ! this.state[ name ],
} );
```
simply set the entire object to [ name ]: ! this.state[ name ]. It did not use the spread operator or other method to keep all other properties for that object intact, and only update the current new value. Somehow that was working.

In the new code, we use the spread operator to copy the settings object, and only update the new values.

```
setSettings( ( prevState ) => ( {
			...prevState,
			[ name ]: ! settings[ name ],
} ) );
```